src/mlir/generator.cc

```diff
@@ -428,7 +368,8 @@ namespace mlir::verona
     return builder.create<LLVM::AddressOfOp>(builder.getUnknownLoc(), global);
   }
 
-  Value MLIRGenerator::Arithmetic(Location loc, StringRef name, Value ops)
+  Value
+  MLIRGenerator::Arithmetic(Location loc, StringRef name, Value ops, Type retTy)
   {
     // FIXME: We already converted U32 to i32 so this "works". But we need
```

Comment:
Is this comment stale now?

Reply:
No, that's not about unsigned vs. signed, just that the classes called `U32` and `I32` etc are converted to MLIR native types very early. This is only a problem if we end up with dynamic dispatch (ex. `a.+(b)`) without knowing the type of `a`. I'm not sure all the cases, but I imagine a union `I32 | F32` where both support `+` would be such a case.

Right now, because we can't implement arithmetic in Verona yet (need the planned changes in the AST for that), it should be fine.